### ISSUE TITLE
When resuming after pause, few chunks remain pending

### ISSUE THREAD

**Issue by muradm** [flowjs/flow.js#108]
Once upload is started, if user pauses it and then resumes, 3 (actually number of simultaneousUploads) chunks remain stale in pending state. Pieces reach the server, but when after resume all other parts are finished, they remain pending.

Looking at the code, the FlowFile.abort method does something wrong?  

```
    /**
     * Abort current upload
     * @function
     */
    abort: function (reset) {
      this.currentSpeed = 0;
      this.averageSpeed = 0;
//      var chunks = this.chunks;
      if (reset) {
        this.chunks = [];
      }

      // NOTE: once below loop commented out, everything seems to be working
      //each(chunks, function (c) {

      // NOTE: ok, here it supposed to cancel the chunks in the wire. but does it really matters?
      // NOTE: if chunk is left in the wire, it is probably will reach server, let it finish
      // NOTE: if chunk still not entered the wire (xhr.send not called yet), check if upload is not paused
      // NOTE: right before send once more.
      // NOTE: otherwise, in most cases it is really meaningless to explicitly canceling the chunks
      //  if (c.status() === 'uploading') {
      //    c.abort();

      // NOTE: why? :)
      //    this.flowObj.uploadNextChunk();
      //  }
      //}, this);
    },
```

**Comment by AidasK** [comment]
If file is stopped/paused, we have stop every chunk upload also. This is how file upload and resume works, it must be instant.

` this.flowObj.uploadNextChunk();` - this is the tricky part. For every stopped chunk, we continue to upload other chunks in a queue if they are not paused. Without this, one paused file would pause upload of other files too.

**Comment by Veabers** [comment]
I found the same problem. The way I saw the issue is that the chunk.preprocessState is set to 1 once the preprocessing is complete. Then the chunk is paused the state is back at pending, but the preprocessState is still 1. I think that is correct, because it prevents the preprocess code from running again.

I fixed the issue by removing the check for chunk.preprocessState === 0 when looking for pending chunks.
